### PR TITLE
Add logout page that directs users to click New Identity button in Tor browser

### LINKED ISSUE #4952: Update UI/docs consistent with "New Identity" behavior change in Tor Browser 9.0
The "New Identity" behavior has changed once again in Tor Browser 9.0. There's now a dedicated button for it:
New identity button

So we have to update these instructions:

> Thank you for exiting your session! Please select "New Identity" from the onion button in the Tor browser's toolbar to clear all history of your SecureDrop usage from this device.

And also the relevant portion in the [source guide](https://docs.securedrop.org/en/release-1.1.0/source.html).

### PR DESCRIPTION
**Pull request by rocodes**
- Add page after logout that directs users to click the New Identity button in Tor browser to complete their session
- Amend session timeout message to reflect new "click new identity" CTA 
- Amend functional test to reflect changes in logout behaviour 

## Status

Ready for Review

## Description of Changes

Closes #4952.

Changes proposed in this pull request:

Additional "logout" route that directs users to click the New Identity button in Tor browser.
Amend functional tests to reflect behaviour of new logout page and modified wording on session timeout.

## Testing

- Scenario 1: Log in to source interface, perform any actions, click "logout". Ensure you are logged out and directed to the new "click New Identity" CTA page. 
- Scenario 2: Navigate to $source_interface_url/logout without logging in, ensure you are immediately redirected to the main source interface page
- (optional Scenario 3: Log in and wait for session timeout, or run session timeout functional test. Ensure you see the "click New Identity" CTA in the session timeout flashed message).
- Visual code review of `source_app/main.py`, `tests/functional/source_nagivation_steps.py`, `source_templates/logout.html`, `source_templates/session_timeout.html` to ensure changes are satisfactory
- Run functional tests

## Deployment

no special considerations

### If you made changes to the server application code:

- [x] Linting (`make lint`) and tests (`make test`) pass in the development container

### CODE REVIEW

**Comment by ninavizz** [comment]
I can haz screenie (using Nicholas' nomenclature for `screenshot`)?

**Comment by rocodes** [comment]
Note that it's pretty much just a placeholder @ninavizz pending your suggestions

Screenshot_2020-02-05 SecureDrop Protecting Journalists and Sources

File: securedrop/source_templates/session_timeout.html [hunk 1]
```diff
@@ -3,6 +3,6 @@
     <img src="{{ url_for('static', filename='i/hand_with_fingerprint.png') }}">
   </div>
   <div class="message"><strong>{{ gettext('Important!') }}</strong><br>
-  <p>{{ gettext('Your session timed out due to inactivity. Please login again if you want to continue using SecureDrop, or select "New Identity" from the onion button in the Tor browser\'s toolbar to clear all history of your SecureDrop usage from this device. If you are not using Tor Browser, restart your browser.') }}</p>
+    <p>{{ gettext('You were logged out due to inactivity. Click the <img src={icon} alt="broom icon" width="16" height="16">&nbsp;<strong>New Identity</strong> button in your Tor browser\'s toolbar. This will clear your Tor browser activity data on this device.').format(icon=url_for('static', filename='i/torbroom-black.png'))  }}</p>
```
Comment (eloquence) [inline securedrop/source_templates/session_timeout.html:6]: "button in your Tor browser's toolbar" -> "button in your Tor browser's toolbar before moving on" per language in https://github.com/freedomofpress/securedrop-ux/issues/94#issuecomment-589456323

**Comment by eloquence** [comment]
All scenarios look good to me in functional testing. A couple of design notes:

## Session timeout scenario
timeout-message

Nina's mockup switches this to a new message bubble style we don't have yet (compare for visual, not wording):

Session expiry mockup

That's a bigger lift and I'm OK with tracking that as a follow-up iteration. @ninavizz I'm curious if there any fixes we can make more quickly that you'd be comfortable with, e.g., removing the hand of god from the message.

## Logout scenario

You have some excessive whitespace here below the headline; just removing that `<br>` may fix:
Screen Shot 2020-04-14 at 15 48 49-fullpage

Compare with the mockup (for visual comparison, not wording):
logout mockup

**Comment by eloquence** [comment]
I'll also note that Nina's design has the "Log in" button back on the Logout page, that's probably something we want to implement, to ensure that a user who accidentally logged out can easily find their way back in (not everyone will understand they can click on the logo).

**Comment by ninavizz** [comment]
I love @eloquence's suggestion to more simply remove the `hand-o-god` from the existing messaging style, if an iterative scooch towards the new style w/o a full style update, is sought. Getting that broom in there and the visual focus, is my primary priority... since the naming of that button is just not intuitive. Thank you, Ro!

**Comment by rocodes** [comment]
fyi for @ninavizz: 

Screenshot_2020-04-16 SecureDrop Protecting Journalists and Sources(1)
Screenshot_2020-04-16 SecureDrop Protecting Journalists and Sources

Someone who is more of a css wizard than I am may have opinions about some of this styling. I have noticed the `stop-bang` doesn't render properly in Chrome (it's squished), but this may not matter since this is a TB-only context (at least in an ideal world)

File: securedrop/source_templates/logout.html [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+  <a href="{{ url_for('main.login') }}" class="btn pull-right"> {{ gettext('LOG IN') }}
+  </a>
+  <br>
+<div>
+  <br>
+  <h1>{{ gettext('One more thing...') }}</h1>
+  <p id="click-new-identity-tor"> {{ gettext('Click the <img src={icon} alt="broom icon" width="16" height="16">&nbsp;<strong>New Identity</strong> button in your Tor browser\'s toolbar. This will clear your Tor browser activity data on this device.').format(icon=url_for('static', filename='i/torbroom-black.png'))  }}</p>
+</div>
+</div>
+{% endblock %}
```
Comment (eloquence) [inline securedrop/source_templates/logout.html:13]: Use of `<div>`s for logical grouping is partially a matter of taste, so leaving up to implementer and final reviewer how you want to group it, but just noting that the most compact version of this template is this:

```html
<a href="{{ url_for('main.login') }}" class="btn pull-right"> {{ gettext('LOG IN') }}</a>
<br class="clearfix">
<h1>{{ gettext('One more thing...') }}</h1>
<p id="click-new-identity-tor">...</p>
```

Regardless of how you want to employ `<div>`s, you can get rid of the `class=""` specification in line 3.

File: securedrop/sass/modules/_flash.sass [hunk 3]
```diff
@@ -3,9 +3,9 @@
     display: flex
     flex-flow: row nowrap
     align-items: flex-start
-    border-radius: 10px
-    padding: 10px
+    border-radius: 2px
     margin: 10px
+    padding: 10px 16px
```
Comment (eloquence) [inline securedrop/sass/modules/_flash.sass:8]: This doesn't quite have the intended effect; it adds a bit more padding than needed (note the amount of whitespace around the icon compared with Nina's mockup), and also changes the styling other flash messages. Instead, I would leave this value at `10px`, and add `padding-left:16px` or similar to the `.important` specification.

File: securedrop/sass/modules/_flash.sass [hunk 4]
```diff
@@ -38,11 +38,25 @@
...
     &.important
-      border: 1px solid #EBDCEB
-      background: #FDFAFD
+      border-top: 1px solid #ece6e7
+      border-left: 6px solid $color_urgent_coral
+      border-bottom: 1px solid $color_urgent_coral
+      border-right: 1px solid #ece6e7
+      background-color: #FDFDFD
+      color: #383838
+      align-items: center
```
Comment (eloquence) [inline securedrop/sass/modules/_flash.sass:52]: Consider adding a bit of `margin-bottom` here to give some breathing room (see the spacing between message and logo in Nina's mockup); I'd go with ~25px for starters.

File: securedrop/source_templates/logout.html (hunk 2 above)
Comment (rocodes) [reply securedrop/source_templates/logout.html:13]: oops- that was a remnant of something else, should not have made it in there. thank you for catching!

File: securedrop/sass/modules/_flash.sass [hunk 5]
```diff
@@ -38,11 +38,25 @@
...
+
+      &:dir(rtl)
+        text-align: right
+        border-top: 1px solid #ece6e7
+        border-left: 1px solid #ece6e7
+        border-bottom: 1px solid $color_urgent_coral
+        border-right: 6px solid $color_urgent_coral
```
Comment (eloquence) [inline securedrop/sass/modules/_flash.sass:59]: Yay for thinking of RTL support :)

File: securedrop/source_templates/flashed.html [hunk 6]
```diff
@@ -7,6 +7,8 @@
         <img src="{{ url_for('static', filename='i/font-awesome/info-circle-black.png') }}" width="20" height="16">
         {% elif category == 'error' %}
         <img class="pull-left" src="{{ url_for('static', filename='i/font-awesome/exclamation-triangle-black.png') }}" width="20" height="17">
+        {% elif category == 'important' %}
+        <img class="pull-left" src="{{ url_for('static', filename='i/bang-stop.png') }}" width="22" height="22">
```
Comment (eloquence) [inline securedrop/source_templates/flashed.html:11]: Due to the flex layout, I don't believe you need the `pull-left` here; it works fine without it in my testing.

Comment (rocodes) [reply securedrop/source_templates/flashed.html]: removed in 0acdd39

File: securedrop/sass/modules/_flash.sass (hunk 4 above)
Comment (rocodes) [reply securedrop/sass/modules/_flash.sass:56]: added margin and moved styling changes from .flash to .flash.important in 0acdd39

File: securedrop/sass/modules/_flash.sass (hunk 3 above)
Comment (rocodes) [reply securedrop/sass/modules/_flash.sass]: fixed in 0acdd39; was worried about internationalization so I have a rtl version as well.

File: securedrop/source_templates/logout.html (hunk 2 above)
Comment (rocodes) [reply securedrop/source_templates/logout.html:8]: fixed in 0acdd39

**Comment by rocodes** [comment]
@zenmonkeykstop added apparmor changes and squashed commits.

File: securedrop/sass/modules/_flash.sass (hunk 4 above)
Comment (rocodes) [reply securedrop/sass/modules/_flash.sass:56]: ✔️ 0acdd39

File: securedrop/source_templates/logout.html (hunk 2 above)
Comment (rocodes) [reply securedrop/source_templates/logout.html:8]: updated in 0acdd39

**Comment by eloquence** [comment]
All good from my end, thanks much for the quick changes @rocode! Have only looked at the design, HTML/CSS, and basic functioning so far, so still needs a full code review from @zenmonkeykstop.

**Comment by zenmonkeykstop** [comment]
Tested in staging env:
```
make build-debs
make staging
```
- [x] no 500 errors on new logout page, no errors in source error log or kernel log (apparmor is good)
- [x] flash message on first and subsequent submits displays correctly with both rtl and ltr text
- [ ] Timeout flash message not displayed correctly - aspect ratio on icon is incorrect (see screenshot)
Screenshot from 2020-04-17 10-23-32

**Comment by rocodes** [comment]
I have seen what @zenmonkeykstop found (squished icon) in Chrome, but not FF. Investigating

**Comment by zenmonkeykstop** [comment]
Confirmed that the icon thing is present on current `develop` as well. It only happens on first use of the flash message icon - when the image is in local cache, it displays correctly.

**Pull request merged**